src/App.tsx

```diff
@@ -1,12 +1,71 @@
 import React from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[];
+  errorMessage: string;
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+    errorMessage: '',
+  };
+
+  loadedGoods = async (goodsLoader: () => Promise<Good[]>) => {
```

Comment:
функция должна называться действием
loadedGoods - загруженныеТовары
loadGoods - загрузитьТовары
```suggestion
  loadGoods/ = async (goodsLoader: () => Promise<Good[]>) => {
```